capellambse_context_diagrams/collectors/element_relation_view.py

```diff
@@ -0,0 +1,183 @@
+                            sources=[item.uuid],
+                            targets=[elem.abstract_type.uuid],
+                            # Add back labels once edge label issue is fixed
+                            # labels=makers.make_label(elem.name),
+                        )
+                    )
+        for cls in self.classes.values():
+            box = self._make_class_box(cls)
+            partition = 0
+            current = cls
+            while current.super and current.super.uuid in self.classes:
+                partition += 1
+                current = current.super
+            box.layoutOptions["partitioning.partition"] = partition
+            self.tree_view_box.children.append(box)
+            if partition > 0:
+                eid = f"__Generalization:{cls.super.uuid}_{cls.uuid}"
+                self.tree_view_box.edges.append(
+                    _elkjs.ELKInputEdge(
+                        id=eid,
+                        sources=[cls.super.uuid],
+                        targets=[cls.uuid],
+                    )
+                )
+
+        right_boxes_n = 0
+        while self.left_boxes_n > right_boxes_n:
+            box = self.left_box.children.pop()
+            self.right_box.children.append(box)
+            n = len(box.children)
+            self.left_boxes_n -= n
+            right_boxes_n += n
+
+        if not self.global_boxes:
+            logger.warning("Nothing to see here")
+            return
+        self.data.children.extend(
+            [self.left_box, self.tree_view_box, self.right_box]
+        )
+
+    def _make_class_box(self, cls: information.Class) -> _elkjs.ELKInputChild:
+        box = makers.make_box(
+            cls, layout_options=makers.DEFAULT_LABEL_LAYOUT_OPTIONS
+        )
+        properties = [
+            _elkjs.ELKInputLabel(
+                text="",
+                layoutOptions=makers.DEFAULT_LABEL_LAYOUT_OPTIONS,
+                width=0,
+                height=0,
+            )
+        ]
+        for prop in cls.properties:
+            if prop.type is None or (
+                isinstance(prop.type, information.Class)
+                and not prop.type.is_primitive
+            ):
+                continue
+
+            text = tree_view._get_property_text(prop)
+            labels = makers.make_label(
+                text,
+                icon=(makers.ICON_WIDTH, 0),
+                layout_options=tree_view.DATA_TYPE_LABEL_LAYOUT_OPTIONS,
+                max_width=math.inf,
+            )
+            properties.extend(labels)
+
+        box.labels.extend(properties)
+        box.width, box.height = makers.calculate_height_and_width(properties)
+
+        return box
+
+
+def collector(
+    diagram: context.ElementRelationViewDiagram, params: dict[str, t.Any]
+) -> _elkjs.ELKInputData:
+    processor = ElementRelationProcessor(diagram, params=params)
+    processor.process()
+    return processor.data
```

Comment:
Instead of processor I would name it Collector. But just from seeing that there isn't anything else happening in this function, or everything is handled by the ElementRelationCollector, why not going for the following:

```suggestion
    return ElementRelationProcessor(diagram, params=params)()
```